url3426a97ee4_rtreffer_minidns_tgz-pJ8-de_measite_minidns_minidnsrepl_MiniDnsReplJ8/src/de/measite/minidns/DNSMessage.java

```diff
@@ -0,0 +1,1133 @@
+                return null;
+            }
+            return INVERSE_LUT[value];
+        }
+
+    }
+
+    /**
+     * The DNS message id.
+     */
+    public final int id;
+
+    /**
+     * The DNS message opcode.
+     */
+    public final OPCODE opcode;
+
+    /**
+     * The response code of this dns message.
+     */
+    public final RESPONSE_CODE responseCode;
+
+    /**
+     * The QR flag of the DNS message header. Note that this will be <code>true</code> if the message is a
+     * <b>response</b> and <code>false</code> if it is a <b>query</b>.
+     * 
+     * @see <a href="https://www.ietf.org/rfc/rfc1035.txt">RFC 1035 § 4.1.1</a>
+     */
+    public final boolean qr;
+
+    /**
+     * True if this is a authorative response.
+     */
+    public final boolean authoritativeAnswer;
+
+    /**
+     * True if message is truncated. Then TCP should be used.
+     */
+    public final boolean truncated;
+
+    /**
+     * True if the server should recurse.
+     */
+    public final boolean recursionDesired;
+
+    /**
+     * True if recursion is possible.
+     */
+    public final boolean recursionAvailable;
+
+    /**
+     * True if the server regarded the response as authentic.
+     */
+    public final boolean authenticData;
+
+    /**
+     * True if the server should not perform DNSSEC validation before returning the result.
+     */
+    public final boolean checkingDisabled;
+
+    /**
+     * The question section content. Usually there will be only one question.
+     * <p>
+     * This list is unmodifiable.
+     * </p>
+     */
+    public final List<Question> questions;
+
+    /**
+     * The answers section records. Note that it is not guaranteed that all records found in this section will be direct
+     * answers to the question in the query. If DNSSEC is used, then this section also contains the RRSIG record.
+     * <p>
+     * This list is unmodifiable.
+     * </p>
+     */
+    public final List<Record<? extends Data>> answerSection;
+
+    /**
+     * The Authority Section. Note that it is not guaranteed that this section only contains nameserver records. If DNSSEC is used, then this section could also contain a NSEC(3) record.
+     * <p>
+     * This list is unmodifiable.
+     * </p>
+     */
+    public final List<Record<? extends Data>> authoritySection;
+
+    /**
+     * The additional section. It eventually contains RRs which relate to the query.
+     * <p>
+     * This list is unmodifiable.
+     * </p> 
+     */
+    public final List<Record<? extends Data>> additionalSection;
+
+    public final int optRrPosition;
+
+    /**
+     * The optional but very common EDNS information. Note that this field is lazily populated.
+     *
+     */
+    private EDNS edns;
+
+    /**
+     * The receive timestamp. Set only if this message was created via parse.
+     * This should be used to evaluate TTLs.
+     */
+    public final long receiveTimestamp;
+
+    protected DNSMessage(Builder builder) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 315-321, lines 323-329, lines 331-337, lines 339-370 into separate methods.